src/inference/src/ie_core.cpp

```diff
@@ -701,6 +702,17 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
         }
     }
 
+    void CleanUpProperties(std::string& deviceName, std::map<std::string, std::string>& config) {
+        // auto-batching is not applicable, if there is auto_batch_timeout, delete it
+        if (deviceName.find("BATCH") == std::string::npos) {
+            const auto& batch_timeout_mode = config.find(ov::auto_batch_timeout.name());
+            if (batch_timeout_mode != config.end()) {
+                if (deviceName.find("AUTO") == std::string::npos && deviceName.find("MULTI") == std::string::npos)
```

Comment:
why do we need this condition? should not we remove this config key if `BATCH` is not in the name?

Can this function be merged with `ApplyAutoBatching` ? Since both of them are related to auto batch

Reply:
1. except for batch plugin, auto/multi will need to keep this config key to convey batch related information
2. agree that it's better to merge with ApplyAutoBatching, we were considering maybe a common function be abstract here can be used in future if any other cases of clean up property. If no such clear request now, we can merge into ApplyAutoBatching, since they are batching related operations. @wgzintel